src/transformers/pipelines/image_classification.py

```diff
@@ -156,7 +157,16 @@ def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Imag
             - **label** (`str`) -- The label identified by the model.
             - **score** (`int`) -- The score attributed by the model for that label.
         """
-        return super().__call__(images, **kwargs)
+        # After deprecation of this is completed, remove the default `None` value for `images`
+        if "images" in kwargs:
+            warnings.warn(
+                "The `images` argument has been renamed to `inputs`. In version 5 of Transformers, `images` will no longer be accepted",
+                FutureWarning,
+            )
```

Comment:
I wonder if this is working as expected. Say you have a user script like this:

```py
pipeline = ImageClassificationPipeline(...)
output = pipeline(images=<my-image>, foo="bar")
```

with this PR, the `pipeline` call will fail with something like "requires 1 positional argument, got 0" because `inputs` is not provided.

Reply:
what I would suggest is to have a decorator like this:

```py
def _rename_positional_arg(old_name: str) -> Callable:

    def _decorator(fn: Callable) -> Callable:

        def _inner(*args, **kwargs):
            if old_name in kwargs:
                if len(args) > 0 or "inputs" in kwargs:
                    raise ValueError(f"Cannot pass 'inputs' and '{old_name}' at the same time.")
                warnings.warn("Deprecation message 'you should use inputs instead of {old_name}' ...")
                kwargs["inputs"] = kwargs.pop(old_name)
        return fn(*args, **kwargs)

    return decorator
```

which you can use like this:

```py
...

@_rename_positional_arg(old_name="images")
def __call__(self, inputs: Union[str, List[str], "Image.Image", List["Image.Image"]] = None, **kwargs):
    ...  # here you are guaranteed "inputs" is passed
```

This way, the pipeline will both if:
- inputs are passed as positional args (as done in Inference API)
- inputs are passed as `inputs`
- inputs are passed as `images`

(I did not find a satisfying decorator name but you have the idea^^)

Reply:
Actually, it should work! I set a new default argument value of `None` for inputs, so the pipeline can be called without that argument. However, it will raise an exception if neither `inputs` nor `images` are passed.

Reply:
In other words:

Users are passing "images" as a positional argument: The argument is renamed to `inputs` and they don't notice any change

Users are passing "images" as a kwarg: `inputs` is now `None` but `**kwargs` contains an `"images"` key. The code detects this and sets `inputs = kwargs.pop("images")`, then prints the deprecation warning

Reply:
Ooh, sorry I did not see the added `= None`. You are completely right, forget what I meant above :)